libpius/mailer.py

```diff
@@ -254,36 +256,40 @@ def _send_mail(self, to, msg):
     msg['From'] = self.mail
     if self.address_override:
       msg['To'] = self.address_override
+      env_to = self.address_override
     else:
       msg['To'] = to
+      env_to = [to, self.mail]
     msg['Date'] = formatdate(localtime=True)
 
-    try:
-      smtp = smtplib.SMTP(self.host, self.port)
-      if self.tls:
-        # NOTE WELL: SECURITY IMPORTANT NOTE!
-        # In python 2.6 if you attempt to starttls() and the server doesn't
-        # understand an exception is raised. However before that, it just
-        # carried on # and one could attempt to auth over a plain-text session.
-        # This is BAD!
-        #
-        # So, in order be secure on older pythons we ehlo() and then check the
-        # response before attempting startls.
-        smtp.ehlo()
-        if not smtp.has_extn('STARTTLS'):
-          # Emulate 2.6 behavior
-          raise smtplib.SMTPException('Server does not support STARTTLS')
-        smtp.starttls()
-        # must re-ehlo after STARTTLS
-        smtp.ehlo()
-        # Don't want to send auth information unless we're TLS'd
-        if self.user:
-          smtp.login(self.user, self.password)
-      if self.address_override:
-        env_to = self.address_override
-      else:
-        # BCC the user...
-        env_to = [msg['To'], self.mail]
+    if self.local_mail_dir:
+      if not os.path.isdir(self.local_mail_dir):
+        os.mkdir(self.local_mail_dir)
+      email = open(os.path.join(self.local_mail_dir, msg['To']), 'w')
+      email.write(str(msg))
```

Comment:
the env_to isn't being written here...

Reply:
Good catch but this opens another question. Do we need the copy to yourself to be blind? There is no standard for a BCC header within an email that a sender is supposed to strip before the SMTP submission. We could look into finding a client that does that or add another command to pius to handle sending these emails. If we don't really need the self copy to be blind then we can just add the extra To header to the message and mail senders should send it to both addresses.

Reply:
Well, sendmail's `-t` does this. It reads all address headers from the message (to,cc,bcc), acts on them and strips out BCC from the message.

I just tried postfix'q sendmail compat mode, and it also does this.

So... that's probably sufficient.

I would say that if the mailqueue mode is activated, take any `env_to`s that are not in the standard `to` and add them as `Bcc` in the message.

Reply:
I added a Bcc header and included a helper function to generate the to_addrs parameter of the send_mail method. Newer versions of python have a send_message method which will extract the headers from the message. I included it with some comments but I think you're still trying to support Python2.